### DISCUSSION TITLE
Multiple react select async componenet one page using the same promiss

### DISCUSSION THREAD

**Discussion by cubecleveland** [JedWatson/react-select#4714] [Q&A]
I have many Async react select on the same page .
Is there any way to have them all share the same promise for loading the data ? so it wont have to cal the api many times on render ?

**Comment by ebonow** [comment]
@cubecleveland AsyncSelect has a prop called `defaultOptions` which should allow you to make one request to your api and then pass in the results mapped to options and passed in as the `defaultOptions` prop into each Select.

**Comment by cubecleveland** [comment]
Thanks!

Does that mean i have to create 2 promise functions ? one to initially
fetched and 1 to pass to the load options ?

As of now im passing no deafultOption so i assume the component just uses
the same fecthing function to fetch on load ?




CUBE Workspace
Corporate: 6240 Som Center Rd #130
Solon, OH 44139
T: 216.245.9629

**Reply by ebonow** [reply]
Pass in the defaultOptions prop as an array of Options when your api call is returned. The state change should trigger an update so that should work.

If not, you can always make the loadOptions look at the inputText and if it is empty, return the defaultOptions otherwise call the api. Looking at the inputText is also a good way to set a minimum length before calling the api.

```jsx 
const [ defaultOptions, setDefaultOptions ] = useState([]);

useEffect(() => {
  (callYourApi().then(resp => {
    const options = mapToOptions(resp);
    setDefaultOptions(options);
  }))();
}, []);
    

return (
  <>
    <AsyncSelect name="Select1" defaultOptions={defaultOptions} {...otherProps} />
    <AsyncSelect name="Select2" defaultOptions={defaultOptions} {...otherProps} />
    <AsyncSelect name="Select3" defaultOptions={defaultOptions} {...otherProps} />
  </>
)};
```
